src/pages/index.tsx

```diff
@@ -7,10 +7,15 @@ import { useState } from "react";
 import Register from "../components/Register";
 import Navbar from "../components/Navbar";
 import Footer from "../components/Footer";
+import { useSelector, useDispatch } from "react-redux";
+import { login, logout } from "../GlobalRedux/store";
+import { AuthState } from "@/types/types";
 
 export default function Page() {
   const [secret, setSecret] = useState<string | null>(null);
   const [publicKey, setPublicKey] = useState<string | null>(null);
+  const dispatch = useDispatch();
+  const credential = useSelector((state: AuthState) => state.walletCredential);
```

Comment:
Does this give just one credential or both public and secretkey? If so, it should be `credentials`.